src/codeGenerator/evaluator.ts

```diff
@@ -353,14 +406,143 @@ export class Evaluator implements ExpressionVisitor {
         }
         return result;  // just return the last result
     }
-    visitOn(_astNode: On, _env: EvaluatorEnv) : EvaluatorData {
-        throw new Error("Method not implemented.");
+
+    // FIXME: lots of switch on type nonsense
```

Comment:
You could use the visitor pattern on the triggers

Reply:
OTOH I'm not sure if mixing Store types into EvaluatorData is really the best idea either, since these aren't types that expressions are supposed to evaluate to.

This is something I'll revisit later, time permitting - got other assignments to look at too.

Reply:
Maybe a separate visitor - not one that produces an EvaluatorData